tests/queries/0_stateless/02233_interpolate_1.sql

```diff
@@ -0,0 +1,39 @@
+# Test WITH FILL without INTERPOLATE
+SELECT n, source, inter FROM (
+   SELECT toFloat32(number % 10) AS n, 'original' AS source, number as inter FROM numbers(10) WHERE number % 3 = 1
+) ORDER BY n WITH FILL FROM 0 TO 11.51 STEP 0.5;
+
+# Test INTERPOLATE with const
+SELECT n, source, inter FROM (
+   SELECT toFloat32(number % 10) AS n, 'original' AS source, number as inter FROM numbers(10) WHERE number % 3 = 1
+) ORDER BY n WITH FILL FROM 0 TO 11.51 STEP 0.5 INTERPOLATE (inter AS 42);
+
+# Test INTERPOLATE with field value
+SELECT n, source, inter FROM (
+   SELECT toFloat32(number % 10) AS n, 'original' AS source, number as inter FROM numbers(10) WHERE number % 3 = 1
+) ORDER BY n WITH FILL FROM 0 TO 11.51 STEP 0.5 INTERPOLATE (inter AS inter);
+
+# Test INTERPOLATE with expression
+SELECT n, source, inter FROM (
+   SELECT toFloat32(number % 10) AS n, 'original' AS source, number as inter FROM numbers(10) WHERE number % 3 = 1
+) ORDER BY n WITH FILL FROM 0 TO 11.51 STEP 0.5 INTERPOLATE (inter AS inter + 1);
+
+# Test INTERPOLATE with incompatible const - should produce error
+SELECT n, source, inter FROM (
+   SELECT toFloat32(number % 10) AS n, 'original' AS source, number as inter FROM numbers(10) WHERE number % 3 = 1
+) ORDER BY n WITH FILL FROM 0 TO 11.51 STEP 0.5 INTERPOLATE (inter AS 'inter'); -- { serverError 6 }
+
+# Test INTERPOLATE with incompatible expression - should produce error
+SELECT n, source, inter FROM (
+   SELECT toFloat32(number % 10) AS n, 'original' AS source, number as inter FROM numbers(10) WHERE number % 3 = 1
+) ORDER BY n WITH FILL FROM 0 TO 11.51 STEP 0.5 INTERPOLATE (inter AS inter||'inter'); -- { serverError 44 }
+
+# Test INTERPOLATE with column from WITH FILL expression - should produce error
+SELECT n, source, inter FROM (
+   SELECT toFloat32(number % 10) AS n, 'original' AS source, number as inter FROM numbers(10) WHERE number % 3 = 1
+) ORDER BY n WITH FILL FROM 0 TO 11.51 STEP 0.5 INTERPOLATE (n AS n); -- { serverError 475 }
+
```

Comment:
Could you please also add some cases with subqueries which were failing before?
Like
```
SELECT
    source,
    inter
FROM
(
    SELECT
        toFloat32(number % 10) AS n,
        'original' AS source,
        number AS inter,
        number + 1 AS inter2
    FROM numbers(10)
    WHERE (number % 3) = 1
)
ORDER BY n ASC WITH FILL FROM 0 TO 11.51 STEP 0.5
INTERPOLATE ( inter AS inter2 + inter )
```
and
```
SELECT
    n,
    source,
    inter
FROM
(
    SELECT
        n,
        source,
        inter,
        inter2
    FROM
    (
        SELECT
            toFloat32(number % 10) AS n,
            'original' AS source,
            number AS inter,
            number + 1 AS inter2
        FROM numbers(10)
        WHERE (number % 3) = 1
    )
    ORDER BY n ASC WITH FILL FROM 0 TO 11.51 STEP 0.5
    INTERPOLATE ( inter AS inter + inter2 )
)
```

Reply:
done